proposals/required-members.md

```diff
@@ -210,10 +210,42 @@ this proposal should attempt to address it.
 
 ### Effect on nullable analysis
 
-Members that are marked `required` are not required to be initialized to a non-nullable state at the end of a constructor. These members are additionally considered by nullable analysis to
-be maybe-null at the beginning of any constructor in a type, unless chaining to a `this` or `base` constructor that is attributed with `SetsRequiredMembersAttribute`.
+Members that are marked `required` are not required to be initialized to a non-nullable state at the end of a constructor. All `required` members from this type and any base types are considered
+by nullable analysis to be maybe-null at the beginning of any constructor in that type, unless chaining to a `this` or `base` constructor that is attributed with `SetsRequiredMembersAttribute`.
```

Comment:
I think *non-nullable* and *maybe-null* could be replaced with more precise terms, but we don't need to fix that in this PR. For example, if the member's type is an unconstrained type parameter we could have a *maybe-default* state at the start of the constructor and require either a *maybe-null* or *not-null* state at the end of the constructor. (I'll convert this comment to issue and we can resolve separately at some point when we have time.)

Reply:
Do you have suggestions for what those terms should be? I can fix them now.

Reply:
I wish I had introduced compact terms for these in https://github.com/dotnet/csharplang/blob/main/proposals/nullable-constructor-analysis.md#an-alternative-approach. We can start now by saying:

- instead of saying members are *maybe-null* at the beginning, say that they have the *default state* at the beginning.
- instead of saying members must be *non-null* at the end, say that they must have a *valid state* at the end (using *a* to indicate that more than one flow state could be *valid*).

I might go ahead and send a PR to define and use these terms over in that doc as well.